core/src/net/HttpClient.hpp

```diff
@@ -101,6 +101,21 @@ namespace ledger {
             std::shared_ptr<api::ExecutionContext> _context;
             Option<std::shared_ptr<spdlog::logger>> _logger;
 
+            static api::ErrorCode getErrorCode(int32_t statusCode) {
+                if (statusCode >= 200 && statusCode < 300 ) {
+                    return api::ErrorCode::FUTURE_WAS_SUCCESSFULL;
+                }
+                return statusCode >= 500 ? api::ErrorCode::UNABLE_TO_CONNECT_TO_HOST :
+                statusCode >= 400 ? api::ErrorCode::HTTP_ERROR :
+                api::ErrorCode::TOO_MANY_REDIRECT;
+            };
+
+            static bool isHttpError(api::ErrorCode errorCode) {
+                return errorCode == api::ErrorCode::HTTP_ERROR ||
+                errorCode == api::ErrorCode::UNABLE_TO_CONNECT_TO_HOST ||
+                errorCode == api::ErrorCode::TOO_MANY_REDIRECT;
```

Comment:
Ditto

Reply:
Nah. Boolean chaining will yield better code than branching. That’s a correct usage of boolean algebra.

Reply:
Agree. I made a jugement mistake here (I was stuck in ternary expression stuff)